src/common/validators/validateBy.ts

```diff
@@ -1,16 +1,25 @@
-import { FnLike } from '@seedcompany/common';
+import { Type } from '@nestjs/common';
 import {
   registerDecorator,
   ValidationOptions,
   ValidatorConstraintInterface,
 } from 'class-validator';
+import { MergeExclusive } from 'type-fest';
 
-export interface ValidateByOptions {
-  name: string;
-  constraints?: any[];
-  validator: ValidatorConstraintInterface | FnLike;
-  async?: boolean;
-}
+export type ValidateByOptions =
+  | MergeExclusive<
+      {
+        validator: ValidatorConstraintInterface;
+        name: string;
+        async?: boolean;
+        constraints?: any[];
+      },
+      {
+        validator: Type<ValidatorConstraintInterface>;
+        constraints?: any[];
+      }
+    >
+  | Type<ValidatorConstraintInterface>;
```

Comment:
This is just a shortcut
```ts
FooConstraint
// same as
{ validator: FooConstraint }
```
Easier, and the only thing unable to be given is the `constraints` list, which we rarely need and poor DX because untyped.